packages/gitbook/src/components/DocumentView/Expandable/Details.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { useParams } from 'next/navigation';
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+function useHash() {
+    const params = useParams();
+    const [hash, setHash] = React.useState<string>(global.location?.hash?.slice(1));
+    React.useEffect(() => {
+        function updateHash() {
+            setHash(global.location?.hash?.slice(1));
+        }
+        global.addEventListener('hashchange', updateHash);
+        updateHash();
+        return () => global.removeEventListener('hashchange', updateHash);
+    }, [params]);
```

Comment:
Why do we need to have `params` as dependency here since it's not used? If it's for a specific reason, can you please comment it so that we don't forget?

Reply:
Looks like it's documented in useScrollToHash, we should merge the 2

Reply:
Thanks. Looks like I can get it to work with `useScrollToHash` with some tweaks to that hook. Then I just need to check for the active element in Details to know to set the open attribute.